### PR TITLE
refactor: update to odo 2.4.2

### LINKED ISSUE #357: Migrate to odo v2.4.1 / v2.4.2
[kind/bug]

- Remove S2I
- Remove binary and git component types

### PR DESCRIPTION
**Pull request by sbouchet**
## What is the purpose of this change? What does it change?
updating odo to 2.4.x. removing  s2i component support .

## Was the change discussed in an issue?
fixes #357

### CODE REVIEW

**Review by jeffmaury — CHANGES_REQUESTED** [review]
IT compilation errors

File: src/main/java/org/jboss/tools/intellij/openshift/ui/component/CreateComponentDialogStep.java [hunk 1]
```diff
@@ -79,20 +71,11 @@ protected void textChanged(DocumentEvent e) {
...
-                model.setSourceType((ComponentSourceType) sourceTypeComboBox.getSelectedItem());
-                updateState();
-            }
-        });
         contextTextField.getDocument().addDocumentListener(new DocumentAdapter() {
             @Override
             protected void textChanged(DocumentEvent e) {
                 model.setContext(contextTextField.getText());
                 informationLabel.setVisible(model.isProjectHasDevfile() || model.isProjectIsEmpty());
-                componentTypeTree.setEnabled(!model.isProjectHasDevfile());
```
Comment (jeffmaury) [inline src/main/java/org/jboss/tools/intellij/openshift/ui/component/CreateComponentDialogStep.java:93]: We should do the same on the list when local project has devfile, no ?

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Link is not working, should be fixed, see https://github.com/jbosstools/jbosstools-openshift/pull/2043

File: src/main/java/org/jboss/tools/intellij/openshift/ui/component/CreateComponentDialogStep.java (hunk 1 above)
Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/ui/component/CreateComponentDialogStep.java:93]: good catch !

**Comment by jeffmaury** [comment]
Seems we have an encoding issue (maybe Windows related: 

image

**Comment by sbouchet** [comment]
> Seems we have an encoding issue (maybe Windows related:
> 
> image


can you check the catalog list output ? the dialog uses the same values. on RHEL, it's:
```
Odo Devfile Components:
NAME                               DESCRIPTION                                                         REGISTRY
java-jboss-eap-xp                  Java stack with EAP XP 2.0, OpenJDK 11 and Maven 3.5                Red Hat Registry
java-jboss-eap-xp-bootable-jar     Java stack with EAP XP 2.0 in bootable Jar mode, OpenJDK 11 ...     Red Hat Registry
dotnet50                           Stack with .NET 5.0                                                 Default Registry
dotnet60                           Stack with .NET 6.0                                                 Default Registry
dotnetcore31                       Stack with .NET Core 3.1                                            Default Registry
go                                 Stack with the latest Go version                                    Default Registry
java-maven                         Upstream Maven and OpenJDK 11                                       Default Registry
java-openliberty                   Java application Maven-built stack using the Open Liberty ru...     Default Registry
java-openliberty-gradle            Java application Gradle-built stack using the Open Liberty r...     Default Registry
java-quarkus                       Quarkus with Java                                                   Default Registry
java-springboot                    Spring Boot® using Java                                             Default Registry
java-vertx                         Upstream Vert.x using Java                                          Default Registry
java-wildfly                       Upstream WildFly                                                    Default Registry
java-wildfly-bootable-jar          Java stack with WildFly in bootable Jar mode, OpenJDK 11 and...     Default Registry
nodejs                             Stack with Node.js 14                                               Default Registry
nodejs-angular                     Stack with Angular 12                                               Default Registry
nodejs-nextjs                      Stack with Next.js 11                                               Default Registry
nodejs-nuxtjs                      Stack with Nuxt.js 2                                                Default Registry
nodejs-react                       Stack with React 17                                                 Default Registry
nodejs-svelte                      Stack with Svelte 3                                                 Default Registry
nodejs-vue                         Stack with Vue 3                                                    Default Registry
php-laravel                        Stack with Laravel 8                                                Default Registry
python                             Python Stack with Python 3.7                                        Default Registry
python-django                      Python3.7 with Django                                               Default Registry

```

**Review by jeffmaury — APPROVED** [review]
LGTM. Encoding issue is tracked here: https://github.com/redhat-developer/intellij-common/issues/120

**Pull request merged**